pkg/reconciler/ingress/resources/gateway.go

```diff
@@ -94,15 +94,19 @@ func SortServers(servers []*istiov1beta1.Server) []*istiov1beta1.Server {
 	return servers
 }
 
-// MakeIngressGateways creates Gateways for a given Ingress.
-func MakeIngressGateways(ctx context.Context, ing *v1alpha1.Ingress, originSecrets map[string]*corev1.Secret, svcLister corev1listers.ServiceLister) ([]*v1beta1.Gateway, error) {
+// MakeIngressTLSGateways creates Gateways for a given Ingress.
+func MakeIngressTLSGateways(ctx context.Context, ing *v1alpha1.Ingress, ingressTLS []v1alpha1.IngressTLS, originSecrets map[string]*corev1.Secret, svcLister corev1listers.ServiceLister) ([]*v1beta1.Gateway, error) {
+	// No need to create Gateway if there is no related ingress TLS.
```

Comment:
I wonder if it might be more future proof to create a Gateway per ksvc, no matter if TLS is here or not.
This way, it is there, already scoped to the specific hosts. Because otherwise I worried we go from a global gateway to a custom one just because TLS was enabled.

Reply:
I consider the Gateway split as two parts: TLS Gateway split and non-TLS (or HTTP) Gateway Split.

It seems no doubt that we need TLS Gateway split as we don't want to keep reconciling the global Gateway.

For non-TLS Gateway split, I think we need more considerations. Currently we do not provide a way to configure the non-TLS behavior through Kingerss. So all Kingresses have the same HTTP behavior by default (unless users directly change the Gateway). From this perspective, I am a little concerned that splitting non-TLS Gateway may cause performance regression for the default behavior (as Istio pilot now needs merging Gateway).